frontend/src/components/pages/SearchReviews/SearchReviews.tsx

```diff
@@ -0,0 +1,81 @@
+import { Box, Center } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+
+import { Review } from "../../../types/ReviewTypes";
+import LoadingSpinner from "../../common/LoadingSpinner";
+import SearchBox from "../SearchBox";
+import mockReviews from "./mockReviews";
+import ReviewsGrid from "./ReviewsGrid";
+
+/**
+ * The component for the page where the user searches and filters reviews.
+ */
+const SearchReviews = (): React.ReactElement => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [searchText, setSearchText] = useState<string>("");
+  const [genresFilter, setGenresFilter] = useState<string[]>([]);
+  const [ageRangeFilter, setAgeRangeFilter] = useState<number[]>([]); // ageRange[0] is min age, ageRange[1] is max age
+  const [displayedReviews, setDisplayedReviews] = useState<Review[]>([]);
+
+  /** Fetches the search queries from url */
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+    const searchQuery = params.has("search_query")
+      ? params.get("search_query")
+      : "";
+    const genres = params.has("genres") ? params.get("genres")?.split(",") : [];
+    const strAgeRange = params.has("ageRange")
+      ? params.get("ageRange")?.split(",")
+      : [];
+    const intAgeRange = strAgeRange?.map((age) => {
+      return Number(age);
+    });
+    if (searchQuery) {
+      setSearchText(searchQuery);
+    }
+    if (genres) {
+      setGenresFilter(genres);
+    }
+    if (intAgeRange) {
+      setAgeRangeFilter(intAgeRange);
+    }
+  }, []);
+
+  /** Changes url to when search filters are changed and fetches search results  */
+  useEffect(() => {
+    setLoading(true);
+    // todo add age range filter too
+    let newSearchUrl = "/magazine/search_results";
```

Comment:
I would suggest moving this URL generation code into a separate helper function (ie one that takes search text, filters as arguments and returns a generated URL). In the future, we'll have other parts of our code also generating URLs and redirecting to this page, and such a function would be useful then.